pkg/handler/storage/client.go

```diff
@@ -184,6 +200,9 @@ func (c *Client) generateV2(ctx context.Context, organizationID, projectID, regi
 			Tags:        conversion.GenerateTagList(request.Metadata.Tags),
 			Size:        *size,
 			Attachments: generateAttachmentList(request.Spec.Attachments),
+			NFS: &regionv1.NFS{
```

Comment:
The field `.Spec.NFS` is tagged as `,omitempty`, indicating that it's optional. Usually I'd expect that to mean that when creating a Spec value, you should only include .NFS if there is explicitly a value to give it.

I admit I don't really understand what the semantics are supposed to be here, especially because the default value of RootSquash is `true` according to `checkRootSquash` here, but also `,omitempty` in the CRD type, and required in the API spec. What does it mean to omit `.Spec.NFS`? What does it mean to have `NFS: &NFS{}`?

Reply:
Similar to above, we can decide what to do (possibly nothing) about the defaulting here and fix it up if necessary.